File: Selenium/Activities/Activity18.java
package activities;

import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

public class Activity18 {

	public static void main(String[] args) {

		WebDriver driver =  new FirefoxDriver();
		
		driver.get("https://v1.training-support.net/selenium/selects");
		
		System.out.println(driver.getTitle());
		
		WebElement selectElement = driver.findElement(By.id("multi-select"));
        
        Select multiSelect = new Select(selectElement);

        multiSelect.selectByVisibleText("Javascript");
        
        for(int i = 4; i<=6 ; i++) {
            multiSelect.selectByIndex(i);
        }
        
        multiSelect.selectByValue("node");
        
        List<WebElement> selectedOptions = multiSelect.getAllSelectedOptions();
        System.out.println("Selected options are: ");
        for(WebElement option : selectedOptions) {
            System.out.println(option.getText());
        }

        multiSelect.deselectByIndex(5);
       
        selectedOptions = multiSelect.getAllSelectedOptions();
        System.out.println("Selected options are: ");
        for(WebElement option : selectedOptions) {
            System.out.println(option.getText());
        }
        
        driver.close();

	}

}

File: Selenium/Activities/Activity18.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.select import Select

with webdriver.Firefox() as driver:

    driver.get("https://v1.training-support.net/selenium/selects")

    print(driver.title)

    multiselect_list = driver.find_element(By.ID, "multi-select")
    
    multiselect = Select(multiselect_list)

    multiselect.select_by_visible_text("Javascript")
    
    for option in multiselect.all_selected_options:
        print("Options selected are: ", option.text)

    for x in range(4, 6):
        multiselect.select_by_index(x)
    for option in multiselect.all_selected_options:
        print("Options selected are: ", option.text)

    multiselect.select_by_value("node")
    for option in multiselect.all_selected_options:
        print("Options selected are: ", option.text)

    multiselect.deselect_by_index(7)
    for option in multiselect.all_selected_options:
        print("Options selected are: ", option.text)

    driver.close()